zero/client_server/server.py

```diff
@@ -52,8 +52,13 @@ def __init__(
             If any other encoder is used, the client should use the same encoder.
             Implement your own encoder by inheriting from `zero.encoder.Encoder`.
         """
-        self._port = port
+        self._broker = None
+        self._device_comm_channel = None
+        self._pool = None
+        self._device_ipc = None
```

Comment:
will give typing error and I didn't want them to be Optional 🤔

Reply:
Refer to: https://github.com/Ananto30/zero/pull/34#discussion_r1160589539
If you didn't want them Optional... hm, that's fine, we can tackle it by some type annotations in the class body and without giving those attributes actual values. Then a too-early access would throw an `AttributeError` and these attributes would not be optional.

Reply:
There you go! https://github.com/Ananto30/zero/pull/34/commits/4acb077dad98a1b200199dd2e0a0eaa2b190eb3d